lib/puppet-lint/plugins/check_whitespace/no_colon_whitespace_after.rb

```diff
@@ -0,0 +1,25 @@
+# Public: Check the manifest tokens for resource type declaration that has no whitespace after a colon, and record a warning for each instance found.
+# #
+# https://docs.puppet.com/guides/style_guide.html#spacing-indentation-and-whitespace
+PuppetLint.new_check(:no_colon_whitespace_after) do
+  def check
+    tokens.each do |token|
+      next if token.next_token.nil?
+      unless [:WHITESPACE, :NEWLINE].include?(token.next_token.type) 
+        if (token.type == :COLON)
```

Comment:
I think this might perform a little better with the `unless/if` order swapped, especially with large amounts of code to test.

I also think it needs to make sure the previous token is a title to prevent false positives. It's somewhat common to see condensed format JSON strings without spacing surrounding colons.